File: String/58_LengthOfLastWord.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-


class Solution(object):
    def lengthOfLastWord(self, s):
        s = s.strip()
        length = 0
        while length < len(s) and s[-(length + 1)] != " ":
            length += 1

        return length


class Solution_2(object):
    def lengthOfLastWord(self, s):
        return len(s.strip().split(' ')[-1])

"""
""
"are"
"we are teams"
"we are teams    "
"""

File: String/58_LengthOfLastWord.cpp
/*
 * @Author: xuezaigds@gmail.com
 * @Last Modified time: 2016-06-27 13:27:33
 */


class Solution {
public:
    int lengthOfLastWord(string s) {
        int len = 0, tail = s.size() - 1;
        while (tail >= 0 && s[tail] == ' ') tail--;
        while (tail >= 0 && s[tail] != ' ') {
            len++;
            tail--;
        }
        return len;
    }
};

/*
""
"are"
"we are teams"
"we are teams    "
*/
